### ISSUE TITLE
Google Sheets:  Needs Serious Love

### ISSUE THREAD

**Issue by wanderslth** [flexiodata/flexio-web-app#253]
TL;DR.  After connecting, it takes zapier ~1 second to get a list of available sheets; it takes Flex.io ~30 seconds.  To get a list of individual tabs in a sheet, it takes Zapier ~1 second; it takes Flex.io ~26 seconds.   It also takes Flex.io ~23 seconds to load a ~50 row sheet into the pipe -- and import is bad.

* Load time is unreasonable (see below)
* Have replicated the "load time hang" error reported by Aaron Rauch this weekend (see bug #254 )
* Imports are broken -- well, they seem to be broken unless you have a clean table (see bug #255 )

---

**Zapier Google Sheets review:**

1.  Connect to Google Sheets (normal oauth)
2.  Following Dialog appears:

![edit step - zapier 3](https://user-images.githubusercontent.com/25911712/27351307-7b411f5a-55c2-11e7-91b1-4e3b825ce0d8.png)

3. Click drop-down for Sheet list (ran this multiple times, avg load of 1.5 seconds to populate):

![edit step - zapier](https://user-images.githubusercontent.com/25911712/27351251-53d0a882-55c2-11e7-9230-13bc13cc4b52.png)

4.  By default, they load the first 50 sheets they find:

![edit step - zapier 1](https://user-images.githubusercontent.com/25911712/27351255-5536c80a-55c2-11e7-8f08-2d428d7a3169.png)

5.  If you click the 'load more' button (~1-1.5 second refresh), you get something like this:

![edit step - zapier 2](https://user-images.githubusercontent.com/25911712/27351258-563b3178-55c2-11e7-8340-90ff099853c0.png)

6.  List is cached; I deleted my thing and came back to it and had an instant list.

7.  For multi-tab sheets, you get a second drop-down (we use "folders", which is a poor UI, since sheets doesn't have a notion of folders.  What we're saying is "worksheets in a sheet")

8.  The drop-down for the tabs in the worksheet was ~1 second:

![edit step - zapier 4](https://user-images.githubusercontent.com/25911712/27351477-1f612b3e-55c3-11e7-8ac2-bf7c4525818a.png)

---

**Flex.io Google Sheets Review**

1.  Connect to Google Sheets (oauth)

![flex io](https://user-images.githubusercontent.com/25911712/27351617-92e76f14-55c3-11e7-8b29-caac974187f1.png)

2.  Select "Add File" 

![flex io 2](https://user-images.githubusercontent.com/25911712/27351843-4ab934b0-55c4-11e7-9a1f-ced8dd46e0a8.png)

3.  https://www.youtube.com/watch?v=lkvXeMvzb4o

4. **_30 Seconds_** later:

![flex io 1](https://user-images.githubusercontent.com/25911712/27351802-2a4685e8-55c4-11e7-8ecc-6aedfd8b8800.png)

5.  To drill into the sheets, like Zapier above, click on the "Cider Buy 2016" folder.

6.  **_26 Seconds_** later:

![flex io 3](https://user-images.githubusercontent.com/25911712/27352003-d52b3e9a-55c4-11e7-94a5-72678922ed92.png)

7. Go to pip builder and run.  

8.   **_23 Seconds_** later:

![flex io 4](https://user-images.githubusercontent.com/25911712/27352136-454782a6-55c5-11e7-98b5-d1652faab344.png)

9.  Import doesn't bring in the proper spreadsheet (it mangles everythign into a single column); will do a separate bug on this.
